src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -0,0 +1,136 @@
+                var start = -1;
+                for (int i = 0; i < val.Length; i++)
+                {
+                    if (start < 0 && Char.IsDigit(val[i]))
+                    {
+                        start = i;
+                    }
+                    else if (start >= 0 && !Char.IsDigit(val[i]))
+                    {
+                        nums.Add(int.Parse(val.Substring(start, i - start)));
+                        start = -1;
+                    }
+                }
+                if (start >= 0)
+                    nums.Add(int.Parse(val.Substring(start, val.Length - start)));
+
+                return DSColor.ByARGB(nums[3], nums[0], nums[1], nums[2]);
+            }
+            catch
+            {
+                return DSColor.ByARGB(255, 0, 0, 0);
+            }
+        }
+        private string SerializeValue()
+        {
+            return dsColor.ToString();
+        }
+        /// <summary>
+        ///     Store color value when graph is saved.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="context"></param>
+        protected override void SerializeCore(XmlElement element, SaveContext context)
+        {
+            base.SerializeCore(element, context);
+
+            XmlElement color = element.OwnerDocument.CreateElement("dsColor");
+            color.InnerText = SerializeValue();
+            element.AppendChild(color);
+        }
+        /// <summary>
+        ///     Restore stored value and set "scolor" to it.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="context"></param>
+        protected override void DeserializeCore(XmlElement element, SaveContext context)
+        {
+            base.DeserializeCore(element, context);
+
+            var colorNode = element.ChildNodes.Cast<XmlNode>().FirstOrDefault(x => x.Name == "dsColor");
+
+            if (colorNode != null)
+            {
+                dsColor = DeserializeValue(colorNode.InnerText);
+            }
+
+        }
+        /// <summary>
+        ///     AST Output.
+        /// </summary>
+        /// <param name="inputAstNodes"></param>
+        /// <returns></returns>
+        [IsVisibleInDynamoLibrary(false)]
+        public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
+        {
+            var av = AstFactory.BuildIntNode(Convert.ToInt32(dsColor.Alpha));
+            var ar = AstFactory.BuildIntNode(Convert.ToInt32(dsColor.Red));
+            var ag = AstFactory.BuildIntNode(Convert.ToInt32(dsColor.Green));
+            var ab = AstFactory.BuildIntNode(Convert.ToInt32(dsColor.Blue));
+
+            var colorNode = AstFactory.BuildFunctionCall(
+                    new Func<int, int, int, int, DSColor>(DSColor.ByARGB), new List<AssociativeNode> { av, ar, ag, ab });
+
+            return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), colorNode) };
```

Comment:
@Gytaco  does this execute correctly multiple times?

@ke-yu because this assignment has no dependencies on the input nodes (this node has no inputs) do you think there will be any issues executing this AST reliably?